tests/integration/api/controllers/person.spec.js

```diff
@@ -0,0 +1,62 @@
+const utils = require('@utils');
+const placeFactory = require('@factories/cht/contacts/place');
+const personFactory = require('@factories/cht/contacts/person');
+const { getRemoteDataContext, Person, Qualifier } = require('@medic/cht-datasource');
+const { expect } = require('chai');
+const userFactory = require('@factories/cht/users/users');
+
+describe('Person API', () => {
+  const places = utils.deepFreeze(placeFactory.generateHierarchy());
+  const patient = utils.deepFreeze(personFactory.build({
+    parent: {
+      _id: places.get('clinic')._id,
+      parent: {
+        _id: places.get('health_center')._id,
+        parent: {
+          _id: places.get('district_hospital')._id
+        }
+      },
+    },
+    phone: '1234567890',
+    reported_date: '2024-05-24T18:40:34.694Z',
+    role: 'patient',
+    short_name: 'Mary'
+  }));
+  const userNoPerms = utils.deepFreeze(userFactory.build({
+    place: places.get('clinic')._id,
+    roles: ['no_perms']
+  }));
+  const dataContext = getRemoteDataContext(utils.getOrigin());
+
+  before(async () => {
+    await utils.saveDocs([...places.values(), patient]);
+    await utils.createUsers([userNoPerms]);
+  });
+
+  after(async () => {
+    await utils.revertDb([], true);
+    await utils.deleteUsers([userNoPerms]);
+  });
+
+  describe('GET /api/v1/person/:uuid', async () => {
+    const getPerson = Person.v1.get(dataContext);
```

Comment:
@lorerod my initial approach here is to end up with the minimal amount of new integration tests by _combining_ tests for the cht-datasource with tests for the person API controller.  Basically, these tests use cht-datasource with the `remote` data context to call the controller via the `/api/v1/person/:uuid` REST endpoint. Then the api controller code uses cht-datasource with the `local` data context (aka Pouch) to do the data interactions with Couch.  So we end up being about to test the following "integration":

cht-datasource (remote) -> api Person controller -> cht-datasource (local) -> Couch

To me, this seems like a valid way to test everything at the same time without compromising the effectiveness/fidelity of the tests at all.  Using the remote cht-datasource to make the REST calls is functionally not much different than using the `utils.request` function to make the REST calls...  But, IDK, let me know what you think about all this!

Reply:
@jkuester , thank you for clarifying this. I agree with your approach to testing using the remote data context. This is enough for testing this change.